File: php-pecl/new_task.php
<?php
$connection = new AMQPConnection();
$connection->connect();

$exchange = new AMQPExchange($connection, '');

$queue = new AMQPQueue($connection);
$queue->declare('task_queue', AMQP_DURABLE);

$message = implode(' ', array_slice($argv, 1));
if(empty($message)) {
    $message = "Hello World!";
}

$exchange->publish(
    $message, 
    'task_queue', 
    0, 
    array(
        'delivery_mode' => 2 # make message persistent
    )
);

echo " [x] Sent $message\n";

$connection->disconnect();

File: dotnet/NewTask/NewTask.cs
using RabbitMQ.Client;
using System.Text;

var factory = new ConnectionFactory { HostName = "localhost" };
using var connection = await factory.CreateConnectionAsync();
using var channel = await connection.CreateChannelAsync();

await channel.QueueDeclareAsync(queue: "task_queue", durable: true, exclusive: false,
    autoDelete: false, arguments: null);

var message = GetMessage(args);
var body = Encoding.UTF8.GetBytes(message);

var properties = new BasicProperties
{
    Persistent = true
};

await channel.BasicPublishAsync(exchange: string.Empty, routingKey: "task_queue", mandatory: true,
    basicProperties: properties, body: body);
Console.WriteLine($" [x] Sent {message}");

static string GetMessage(string[] args)
{
    return ((args.Length > 0) ? string.Join(" ", args) : "Hello World!");
}
